### ISSUE TITLE
slicing dimension issue

### ISSUE THREAD

**Issue by dkazanc** [DiamondLightSource/httomo#142]
I've got a question, currently [here](https://github.com/DiamondLightSource/httomo/blob/ea11ba1c4fa6991e31138392a25027b45c1141c6/httomo/utils.py#L205) we've got this for slicing dimension
```
    if pattern == Pattern.projection:
        return 1
    elif pattern == Pattern.sinogram:
        return 2
```

The data however is the size `(1871, 2160, 2560)`, `(projections, detY, detX)`

when pattern is projection then the slicing is 1, but it should be 0, isn't it? And 1 for sinogram pattern instead of 2. 

hen in [reslice](https://github.com/DiamondLightSource/httomo/blob/ea11ba1c4fa6991e31138392a25027b45c1141c6/httomo/data/hdf/_utils/reslice.py#L57) function where the slicing dimension is used we see that 1 is deducted everywhere. Is that intentional? To me it looks a bit weird. I will try fixing this in memory branch as I need the **correct** slicing dimension when I work with sections!

**Comment by dkazanc** [comment]
[4ef1401](https://github.com/DiamondLightSource/httomo/pull/106/commits/4ef140103995508b9a1cd67a5d69af872bca71b4)
Fixed it as I see it and tested with in-memory and file-based-reslice. Seems working fine.

**Comment by yousefmoazzam** [comment]
Mmm, I agree that the value of `slice_dim` is unclear, maybe I can shed some light...

If I recall correctly, the reslicing stuff in httomo from its beginning (after Jacob's work and after the Nvidia hackathon in September 2022) was such that the "slice dimension" used _one_-based indexing rather than _zero_-based indexing.

So if `slice_dim=1` then this meant slice in the 0th axis of the numpy array, and if `slice_dim=2` then slice in the 1st axis of the array. It's something that came from Jacob's work I think, since you can see the `slice_dim-1` being used in that code back then https://github.com/jacob720/Load-H5/blob/master/h5_utils/chunk_h5.py#L45-L81

With that being said, I'd be careful to make sure that the data is being resliced correctly with any change to `slice_dim` using zero-based indexing, to make sure things are consistent. For example, in the `reslice()` function https://github.com/DiamondLightSource/httomo/blob/ea11ba1c4fa6991e31138392a25027b45c1141c6/httomo/data/hdf/_utils/reslice.py#L55-L69

we can see that it does stuff like `current_slice_dim-1` and `next_slice_dim-1`, so it's relying on `slice_dim` using one-based indexing, not zero-based indexing.

Conceptually it's totally fine to make the change from one-based indexing to zero-based indexing for `slice_dim`, it's a small change so I've got no issue with that, we just need to be careful to change everything that is relevant to avoid inconsistencies :slightly_smiling_face:

**Comment by dkazanc** [comment]
yeah, thanks! yes slice_dim is all around the place. mpirun currently doesn't work for me. Will do another check.
